headers.js

```diff
@@ -1,119 +1,147 @@
-    'https://onramp.gem.co',
-    // @shapeshiftoss/market-service@2.0.0: https://github.com/shapeshift/lib/blob/1efccc3401eccb3125e1f09b7f829b886b457b89/packages/market-service/src/osmosis/osmosis.ts#L21
-    'https://api-osmosis.imperator.co/tokens/',
-    // captcha
-    'https://api.friendlycaptcha.com/api/v1/',
-    process.env.REACT_APP_ETHEREUM_NODE_URL,
-    process.env.REACT_APP_UNCHAINED_ETHEREUM_HTTP_URL,
-    process.env.REACT_APP_UNCHAINED_ETHEREUM_WS_URL,
-    process.env.REACT_APP_UNCHAINED_BITCOIN_HTTP_URL,
-    process.env.REACT_APP_UNCHAINED_BITCOIN_WS_URL,
-    process.env.REACT_APP_UNCHAINED_COSMOS_HTTP_URL,
-    process.env.REACT_APP_UNCHAINED_COSMOS_WS_URL,
-    // Explicitly whitelist our KeepKey versions file
-    process.env.REACT_APP_KEEPKEY_VERSIONS_URL,
-    process.env.REACT_APP_WALLET_MIGRATION_URL,
-    process.env.REACT_APP_UNCHAINED_OSMOSIS_HTTP_URL,
-    process.env.REACT_APP_UNCHAINED_OSMOSIS_WS_URL,
-  ],
-  'frame-src': ['https://fwd.metamask.io/', 'https://widget.portis.io'],
-  'img-src': [
-    "'self'",
-    'data:',
-    'blob:',
-    'filesystem:',
-    // https://github.com/shapeshift/web/blob/70111acd6236759675cf81ddcd196c31472989a6/src/components/Modals/FiatRamps/utils.ts#L13
-    process.env.REACT_APP_GEM_ASSET_LOGO,
-    'https://assets.coincap.io/assets/icons/',
-    'https://static.coincap.io/assets/icons/',
-    'https://assets.coingecko.com/coins/images/',
-    'https://raw.githack.com/trustwallet/assets/',
-    'https://rawcdn.githack.com/yearn/yearn-assets/',
-    'https://raw.githack.com/yearn/yearn-assets/',
-    'https://assets.yearn.network/tokens/',
-    'https://raw.githubusercontent.com/yearn/yearn-assets/',
-    'https://rawcdn.githack.com/trustwallet/assets/',
-    'https://raw.githubusercontent.com/osmosis-labs/',
-    'https://raw.githack.com/shapeshift/lib/',
-    'https://raw.githubusercontent.com/shapeshift/lib/',
-    'https://raw.githubusercontent.com/cosmostation/',
-    'https://raw.githubusercontent.com/cosmos/chain-registry/',
-  ],
-  'script-src': [
-    "'self'",
-    'blob:',
-    "'unsafe-eval'", //TODO: There are still a couple of libraries we depend on that use eval; notably amqp-ts and google-protobuf.
-    "'unsafe-inline'", //TODO: The only inline code we need is the stub injected by Metamask. We can fix this by including the stub in our own bundle.
-    "'report-sample'",
-  ],
-  'style-src': ["'self'", "'unsafe-inline'", "'report-sample'"],
-  'base-uri': ["'none'"],
-  'object-src': ["'none'"],
-})
-  .map(([k, v]) => `${[k, ...v].join(' ')}`)
-  .join('; ')
+/** @typedef {Record<string, Array<string>>} CspBase */
+/** @typedef {[string, string]} CspEntry */
+
+/**
+ * @param {CspBase} x
+ * @returns {CspEntry[]}
+ */
+function cspToEntries(x) {
+  return Object.entries(x).flatMap(([k, v]) => {
+    return v.map(entry => {
+      /** @type {[string, string]} */
+      const out = [k, entry]
+      return out
+    })
+  })
+}
+
+/**
+ * @param {CspEntry[]} x
```

Comment:
Please add a description of the function.

I'd also like to see some tests to verify algorithm.

Reply:
Done!